File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        a = n
        sum = 0
        prod = 1
        while a > 0:
            sum += a % 10
            prod *= a % 10
            a //= 10
        return n % (sum + prod) == 0
        

File: 3622-check-divisibility-by-digit-sum-and-product/3622-check-divisibility-by-digit-sum-and-product.java
class Solution {
    private int[] digitsum(int n){
        int[] s = new int[2];
        s[0] = n % 10;
        s[1] = n % 10;
        n /= 10;
        while(n > 0){
            s[0] += n % 10;
            s[1] *= n % 10;
            n /= 10;
        }
        return s;
    }
    public boolean checkDivisibility(int n) {
        int[] sum = digitsum(n);
        return n % (sum[0] + sum[1]) == 0;
    }
}